notebooks/implicit_differentiation.ipynb

```diff
@@ -0,0 +1,561 @@
+    "\n",
+    "I'll first define a new function that uses implicit differentiation and then I'll highlight the differences."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "consistent-singing",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Let's go back to our small system in order to compare explicit and implicit differentiation.\n",
+    "N = 7\n",
+    "density = 0.3\n",
+    "\n",
+    "dimension = 2\n",
+    "box_size = quantity.box_size_at_number_density(N,density,dimension)\n",
+    "displacement, shift = space.periodic(box_size) \n",
+    "\n",
+    "key = random.PRNGKey(5)\n",
+    "key, split = random.split(key)\n",
+    "R_init = random.uniform(key, (N,dimension), minval=0.0, maxval=box_size, dtype=f64) \n",
+    "\n",
+    "sigma = jnp.full((N,N), 2.0)\n",
+    "alpha = jnp.full((N,N),2.)\n",
+    "param_dict = {'sigma':sigma,'alpha':alpha}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "mature-discharge",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def implicit_diff(params,R_init,displacement):\n",
+    "    energy_fn = energy.soft_sphere_pair(displacement, **params)\n",
+    "    force_fn = jit(quantity.force(energy_fn))\n",
+    "    \n",
+    "    # wrap force_fn with a lax.stop_gradient to prevent a CustomVJPException\n",
+    "    no_grad_force_fn = jit(lambda x: lax.stop_gradient(force_fn(x)))\n",
+    "    \n",
+    "    # make the dependence on the variables we want to differentiate explicit\n",
+    "    explicit_force_fn = jit(lambda R, p: force_fn(R, **p))\n",
+    "    \n",
+    "    def solver(params, x):\n",
+    "        # params are unused\n",
+    "        del params\n",
+    "        # need to use no_grad_force_fn!\n",
+    "        return run_minimization_while(no_grad_force_fn, x, shift, dt_start=0.001, dt_max=0.005)[0]\n",
+    "    \n",
+    "    decorated_solver = jaxopt.implicit_diff.custom_root(explicit_force_fn)(solver)\n",
+    "    \n",
+    "    R_final = decorated_solver(None,R_init)\n",
+    "    \n",
+    "    # Here we can just use our original energy_fn/force_fn\n",
+    "    return energy_fn(R_final), jnp.amax(jnp.abs(force_fn(R_final)))\n",
+    "\n",
+    "(imp_e, imp_f), imp_g = jax.value_and_grad(implicit_diff,has_aux=True)(param_dict,R_init,displacement)\n",
+    "print('Energy        : ',imp_e)\n",
+    "print('Max_grad_force: ',imp_f)\n",
+    "print('Gradient of the energy:')\n",
+    "print(imp_g['sigma'][0])\n",
+    "print(imp_g['alpha'][0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "biblical-pharmacy",
+   "metadata": {},
+   "source": [
+    "As you can see we do not have to change anything compared to our explicit version when we want to compute gradients with respect to more then one parameter."
```

Comment:
I wasn't sure about this sentence (since some code did need to be changed with respect to the explicit version).  Maybe move the comment below to this text just stating that the results between the implicit solver and the explicit solver agree.